plugin/v2/plugin.go

```diff
@@ -309,7 +358,7 @@ func (p *Plugin) InitSpec(s specs.Spec, libRoot string) (*specs.Spec, error) {
 			m.Source = *mount.Source
 		}
 		if m.Source != "" && m.Type == "bind" {
-			fi, err := os.Lstat(filepath.Join(rootfs, m.Destination)) // TODO: followsymlinks
+			fi, err := os.Lstat(filepath.Join(p.Rootfs, string(os.PathSeparator), m.Destination)) // TODO: followsymlinks
```

Comment:
if you want to prevent `../` then you need to get the `Abs` of the path and compare.  lstat will follow the symlink but i'm not sure what you are trying to do here, its naive to look for a destination like this and i don't know why u have to look for the dest to create the source. 

the dest in the resulting container has to take all the previous mounts into account so i'm unsure that this will do what you want in the end